File: scripts/agentcadia_tools.py
#!/usr/bin/env python3
import argparse
import json
import pathlib
import runpy
import sys


def main():
    if len(sys.argv) < 2 or sys.argv[1] not in {"upload", "download", "eval"}:
        print(
            "Usage: python3 scripts/agentcadia_tools.py <upload|download|eval> [args...]",
            file=sys.stderr,
        )
        sys.exit(2)

    command = sys.argv[1]
    script_dir = pathlib.Path(__file__).resolve().parent
    if command == 'upload':
        target = script_dir / 'upload_agentcadia.py'
    elif command == 'download':
        target = script_dir / 'download_agentcadia.py'
    else:
        target = script_dir / 'eval_agentcadia.py'
    sys.argv = [str(target), *sys.argv[2:]]
    runpy.run_path(str(target), run_name='__main__')


if __name__ == "__main__":
    main()

File: bin/agentcadia-tools.js
#!/usr/bin/env node
const { spawnSync } = require('node:child_process');
const { resolve } = require('node:path');

const script = resolve(__dirname, '../scripts/agentcadia_tools.py');
const args = process.argv.slice(2);
const firstArg = args[0];

if (
  args.length === 0 ||
  firstArg === '--help' ||
  firstArg === '-h' ||
  firstArg === 'help'
) {
  console.log([
    'Agentcadia Tools',
    '',
    'Usage:',
    '  agentcadia-tools <upload|download|eval> [args...]',
    '',
    'Examples:',
    '  agentcadia-tools upload --help',
    '  agentcadia-tools download --help',
    '  agentcadia-tools eval start --help',
    '',
    'This CLI is a thin Node wrapper around scripts/agentcadia_tools.py.',
    'Python 3 must be available as `python3` in PATH.'
  ].join('\n'));
  process.exit(0);
}

const result = spawnSync('python3', [script, ...args], { stdio: 'inherit' });

if (result.error) {
  console.error(`Failed to launch python3: ${result.error.message}`);
  process.exit(1);
}

process.exit(result.status ?? 1);
